### PR TITLE
fix warnings introuduced with ruby 2.7, which warns about backward co…

### PR DESCRIPTION
**Pull request by pruan-rht**
…mpatiblity against ruby 3.0  for example:

> DEPRECATION WARNING: Ruby 2.5+ (required by Rails 6) provides Hash#transform_values natively, so requiring active_support/core_ext/hash/transform_values is no longer necessary. Requiring it will raise LoadError in Rails 6.1. (called from require at /home/pruan/workspace/bushslicer/features/step_definitions/dc.rb:4)
> /home/pruan/workspace/bushslicer/lib/host.rb:532: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call

### CODE REVIEW

**Comment by pruan-rht** [comment]
for reference https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/ . Tested against ruby 2.6.3  it worked

File: features/step_definitions/dc.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 
 require 'active_support/core_ext/hash/slice.rb'
-require 'active_support/core_ext/hash/transform_values.rb'
+#require 'active_support/core_ext/hash/transform_values.rb'
```
Comment (rbbratta) [inline features/step_definitions/dc.rb]: could we just remove the line instead of commenting it out?

Comment (pruan-rht) [reply features/step_definitions/dc.rb]: @rbbratta sure, done.

**Comment by pruan-rht** [comment]
@akostadinov PTAL.  Thanks!

File: features/step_definitions/deployment.rb [hunk 2]
```diff
@@ -1,7 +1,6 @@
 # frozen_string_literal: true
 
 require 'active_support/core_ext/hash/slice.rb'
-require 'active_support/core_ext/hash/transform_values.rb'
```
Comment (akostadinov) [inline features/step_definitions/deployment.rb]: I know this is causing deprecation warning. But we still support running on 2.4. Perhaps require it conditionally if RUBY version is above 2.5?

File: lib/environment_manager.rb [hunk 3]
```diff
@@ -37,7 +37,7 @@ def conf
       raise "no '#{key}' environment configuration found" if ! env_class || env_class.empty?
 
       env_opts[:key] = key # lets have it for a reference
-      return BushSlicer.const_get(env_class).new(env_opts)
+      return BushSlicer.const_get(env_class).new(**env_opts)
```
Comment (akostadinov) [inline lib/environment_manager.rb]: why do we need this change?

File: lib/host.rb [hunk 4]
```diff
@@ -529,9 +529,9 @@ def delete(file, opts={})
       else
         exec_method = :exec
       end
-      public_send(exec_method, "rm #{r} -f -- #{file}", opts)
+      public_send(exec_method, "rm #{r} -f -- #{file}", **opts)
       opts[:quiet] = true
-      res = public_send(exec_method, "ls -d -- #{file}", opts)
+      res = public_send(exec_method, "ls -d -- #{file}", **opts)
```
Comment (akostadinov) [inline lib/host.rb]: You sure we need these changes?

File: features/step_definitions/deployment.rb (hunk 2 above)
Comment (akostadinov) [reply features/step_definitions/deployment.rb]: or shall we stop supporting ruby 2.4? Maybe ask @jianlinliu

File: lib/environment_manager.rb (hunk 3 above)
Comment (pruan-rht) [reply lib/environment_manager.rb]: @akostadinov because w/o the change we will see this warning  `warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call`

File: lib/host.rb (hunk 4 above)
Comment (pruan-rht) [reply lib/host.rb]: @akostadinov yes.  https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/

File: features/step_definitions/deployment.rb (hunk 2 above)
Comment (pruan-rht) [reply features/step_definitions/deployment.rb]: @jianlinliu WDYT about the stop support for ruby 2.4?

Comment (pruan-rht) [reply features/step_definitions/deployment.rb]: @jianlinliu can you please comment on this?

Comment (jianlinliu) [reply features/step_definitions/deployment.rb]: I am not sure what is meaning of support for ruby 2.4. I have no any cucushift test cases.

File: features/step_definitions/dc.rb [hunk 5]
```diff
@@ -150,6 +149,15 @@
 end
 
 Given /^master CA is added to the#{OPT_QUOTED} dc$/ do |name|
+  ensure_admin_tagged
```
Comment (cben) [inline features/step_definitions/dc.rb]: this chunk seems unrelated to this PR?

File: features/step_definitions/deployment.rb (hunk 2 above)
Comment (pruan-rht) [reply features/step_definitions/deployment.rb:4]: @akostadinov @jianlinliu with F32 out now, the default ruby is 2.7.1, can we merge this PR?  Otherwise, the cucumber output will have lots of warnings.  @jianlinliu what it means is that we would update our jenkins slave pods to have ruby > 2.5

File: tools/find_image.rb [hunk 6]
```diff
@@ -32,14 +32,14 @@ def run
...
-        doc = Nokogiri::HTML(open(options.url).read)
+        doc = Nokogiri::HTML(URI.open(options.url).read)
         links = doc.xpath("//a[contains(text(), '#{options.build_type}')]")
 
         target_links = links.map { |l|  options.url + l.attributes['href'].value }
         threads = []
         target_links.each do |link|
           threads << Thread.new(link) do |i|
-            doc = Nokogiri::HTML(open(link).read)
+            doc = Nokogiri::HTML(URI.open(link).read)
```
Comment (akostadinov) [inline tools/find_image.rb:42]: @pruan-rht , Ithink the other fixes are already merged, do you want to submit only changes to this file?
